packages/firestore/src/index/ordered_code_writer.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law | agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES | CONDITIONS OF ANY KIND, either express | implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { fail } from '../util/assert';
+import { ByteString } from '../util/byte_string';
+
+export class OrderedCodeWriter {
```

Comment:
optional: copy comment over and add a TODO that this section will be implemented in the future, rather than intentionally not implemented.

Reply:
This class is already in review: https://github.com/firebase/firebase-js-sdk/pull/5821